lib/runner.js

```diff
@@ -75,23 +85,23 @@ Runner.prototype.serial = function (cb) {
 Runner.prototype.run = function (cb) {
 	var concurrent = this.tests.concurrent;
 	var serial = this.tests.serial;
-
-	if (serial.length > 0 && concurrent.length === 0) {
-		this.serial(this.end.bind(this, cb));
-		return;
-	}
-
-	if (serial.length === 0 && concurrent.length > 0) {
-		this.concurrent(this.end.bind(this, cb));
-		return;
-	}
-
-	if (serial.length > 0 && concurrent.length > 0) {
-		this.serial(this.concurrent.bind(this, this.end.bind(this, cb)));
-		return;
-	}
-
-	this.end(cb);
+	var before = this.tests.before;
+	var after = this.tests.after;
+
+	// TODO: refactor this bullshit
```

Comment:
Yeah, I thought about converting `this.serial` and `this.concurrent` to return promises, but decided postpone that. Wanted to hear feedback on a PR and thought that you might not like too much changes at once, haha